tests/unit_tests/c_wrappers/integration/test_field_parameter.py

```diff
@@ -594,3 +594,116 @@ def test_parameter_update(tmpdir):
         assert not np.equal(pp1, pp2).all()
         assert not np.equal(pp2, pp3).all()
         assert not np.equal(pp3, pp4).all()
+
+
+def test_config_node_meta_information(storage, tmpdir):
+    """
+    Populate nodes GEN_DATA, GEN_KW, FIELD, SURFACE & SUMMARY in configuration
+    Verify that properties stored for these nodes are correct
+    """
+
+    with tmpdir.as_cwd():
+        config = dedent(
+            """
+        NUM_REALIZATIONS 1
+        GRID MY_EGRID.EGRID
+        SURFACE TOP     OUTPUT_FILE:surf.irap   INIT_FILES:Surfaces/surf%d.irap   BASE_SURFACE:Surfaces/surf0.irap
+        SURFACE BOTTOM  OUTPUT_FILE:surf.wrap   INIT_FILES:Surfaces/surf%d.wrap   BASE_SURFACE:Surfaces/surf0.wrap  FORWARD_INIT:True
+
+        GEN_DATA ABC  RESULT_FILE:SimulatedABC_%d.txt  INPUT_FORMAT:ASCII   REPORT_STEPS:0
+        GEN_DATA DEF  RESULT_FILE:SimulatedDEF_%d.txt  INPUT_FORMAT:ASCII   REPORT_STEPS:0
+
+        GEN_KW KW_NAME  template.txt kw.txt prior.txt INIT_FILES:custom_param%d.txt
+
+        ECLBASE eclipse/model/MY_VERY_OWN_OIL_FIELD-%d
+        SUMMARY WOPR:MY_WELL
+        SUMMARY WOPR:MY_BASIN
+
+        FIELD MY_PARAM PARAMETER my_param.grdecl INIT_FILES:my_param_%d.grdecl MIN:0.5 MAX:0.8
+        FIELD MY_PARAM2 PARAMETER my_param.grdecl INIT_FILES:my_param_%d.grdecl MIN:0.5 MAX:0.8 FORWARD_INIT:True
+        """  # pylint: disable=line-too-long  # noqa: E501
+        )
+
+        expect_surface = Surface(
+            nx=1, ny=3, xinc=1, yinc=1, xstart=1, ystart=1, angle=0
+        )
+        expect_surface.write("Surfaces/surf0.irap")
+        expect_surface_top = Surface(
+            nx=1, ny=3, xinc=1, yinc=1, xstart=1, ystart=1, angle=0
+        )
+        expect_surface_top.write("Surfaces/surf0.wrap")
+
+        with open("template.txt", mode="w", encoding="utf-8") as fh:
+            fh.writelines("MY_KEYWORD <MY_KEYWORD>")
+        with open("prior.txt", mode="w", encoding="utf-8") as fh:
+            fh.writelines("MY_KEYWORD NORMAL 0 1")
+        with open("custom_param0.txt", mode="w", encoding="utf-8") as fh:
+            fh.writelines("MY_KEYWORD 1.31")
+
+        with open("config.ert", "w", encoding="utf-8") as fh:
+            fh.writelines(config)
+
+        grid = xtgeo.create_box_grid(dimension=(10, 10, 1))
+        grid.to_file("MY_EGRID.EGRID", "egrid")
+
+        buffer = np.random.random_sample(100)
+        buffer[56] = 0.001
+        buffer[34] = 1.001
+        write_grid_property(
+            "MY_PARAM", grid, "my_param_0.grdecl", "grdecl", (10, 10, 1), buffer
+        )
+
+        ert, _ = create_runpath(storage, "config.ert", [True])
```

Comment:
Could be an idea to just have an intermediate variable `ensemble_config = ert.ensembleConfig()` here.